test/e2e/flowaggregator_test.go

```diff
@@ -1865,6 +1879,68 @@ func getAndCheckFlowAggregatorMetrics(t *testing.T, data *TestData) error {
 	return nil
 }
 
+func testL7FlowExporterController(t *testing.T, data *TestData, isIPv6 bool) {
+	skipIfFeatureDisabled(t, features.L7FlowExporter, true, false)
+	nodeName := nodeName(0)
+	_, serverIPs, cleanupFunc := createAndWaitForPod(t, data, data.createNginxPodOnNode, "l7flowexportertestpodserver", nodeName, data.testNamespace, false)
+	defer cleanupFunc()
+
+	clientPodName := "l7flowexportertestpodclient"
+	clientPodLabels := map[string]string{"flowexportertest": "l7"}
+	clientPodAnnotations := map[string]string{antreaagenttypes.L7FlowExporterAnnotationKey: "both"}
+	require.NoError(t, NewPodBuilder(clientPodName, data.testNamespace, toolboxImage).OnNode(nodeName).WithContainerName("l7flowexporter").WithLabels(clientPodLabels).WithAnnotations(clientPodAnnotations).Create(data))
+	clientPodIPs, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace)
+	require.NoErrorf(t, err, "Error when waiting for IP for Pod '%s': %v", clientPodName, err)
+	defer deletePodWrapper(t, data, data.testNamespace, clientPodName)
+
+	testFlow1 := testFlow{
+		srcPodName: clientPodName,
+	}
+	var cmd []string
+	if !isIPv6 {
+		testFlow1.srcIP = clientPodIPs.IPv4.String()
+		testFlow1.dstIP = serverIPs.IPv4.String()
+		cmd = []string{
+			"curl",
+			fmt.Sprintf("http://%s:%d", serverIPs.IPv4.String(), serverPodPort),
+		}
+	} else {
+		testFlow1.srcIP = clientPodIPs.IPv6.String()
+		testFlow1.dstIP = serverIPs.IPv6.String()
+		cmd = []string{
+			"curl",
+			"-6",
+			fmt.Sprintf("http://[%s]:%d", serverIPs.IPv6.String(), serverPodPort),
+		}
+	}
+	stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, testFlow1.srcPodName, "l7flowexporter", cmd)
+	require.NoErrorf(t, err, "Error when running curl command, stdout: %s, stderr: %s", stdout, stderr)
+	_, recordSlices := getCollectorOutput(t, testFlow1.srcIP, testFlow1.dstIP, "", false, false, isIPv6, data, "")
+	for _, record := range recordSlices {
+		if strings.Contains(record, testFlow1.srcIP) && strings.Contains(record, testFlow1.dstIP) {
```

Comment:
We don't need this if here as the records in recordsSlices should match the srcIP/dstIP/srcPort already.